### PR TITLE
added visit_topdown methods to Visitor classes

### PR DESCRIPTION
**Pull request by llpinokio**
I think it would be useful have visit_top_down methods in visitors.

### CODE REVIEW

**Comment by erezsh** [comment]
Looks good, but please include a test that shows that it works. You can modify the test used for bottom-up visits

**Comment by llpinokio** [comment]
@erezsh sure!, but where can i find it ?, i couldn't find in the tests or example folders...
i created this one

setup:
```python
from lark.tree import Tree
from lark.visitors import Visitor,Visitor_Recursive

class Printer(Visitor):
	def __default__(self,tree):
		print(tree.data)
class Printer_Recursive(Visitor_Recursive):
	def __default__(self,tree):
		print(tree.data)
tree=Tree("a",[
	Tree("b",[
		Tree("c",[]),
		Tree("d",[])
	]),
	Tree("e",[
		Tree("f",[]),
		Tree("g",[])
	])
])
```

iterative
```python
 Printer().visit_top_down(tree)
```
> a
b
c
d
e
f
g

recursive
```python
 Printer_Recursive().visit_top_down(tree)
```
> a
b
c
d
e
f
g

bottom-up
```python
Printer().visit(tree)
```
>c
d
b
f
g
e
a

**Comment by evandrocoan** [comment]
I think they are around here: https://github.com/lark-parser/lark/blob/master/tests/test_trees.py#L31 but I could not find the bottom up tests.

**Comment by llpinokio** [comment]
added on `tests/test_trees.py`

File: tests/test_trees.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 import functools
 
 from lark.tree import Tree
-from lark.visitors import Transformer, Interpreter, visit_children_decor, v_args, Discard
+from lark.visitors import Visitor,Visitor_Recursive,Transformer, Interpreter, visit_children_decor, v_args, Discard
```
Comment (evandrocoan) [inline tests/test_trees.py]:
```python
Visitor,Visitor_Recursive,Transformer, Interpreter
-->
Visitor, Visitor_Recursive, Transformer, Interpreter
```

File: lark/visitors.py [hunk 2]
```diff
@@ -198,12 +203,21 @@ def visit(self, tree):
...
+        for child in tree.children:
+            if isinstance(child, Tree):
+                self.visit_top_down(child)
+        
         return tree
 
 
 
+
 def visit_children_decor(func):
```
Comment (evandrocoan) [inline lark/visitors.py]: Why another new line?

**Comment by erezsh** [comment]
Looks great! Just fix the formatting and I'll merge it in.

**Comment by llpinokio** [comment]
@erezsh is it okay now ?

**Comment by erezsh** [comment]
Yes, thanks!

**Pull request merged**